packages/language/test/lsp/definition-requests.test.ts

```diff
@@ -0,0 +1,170 @@
+import { describe, it, expect, beforeEach, afterEach } from "vitest";
+import { URI } from "../../src/utils/uri";
+import { TextDocument } from "vscode-languageserver-textdocument";
+import { TextDocuments } from "../../src/language-server/text-documents";
+import { replaceIndices } from "../utils";
+import { VirtualFileSystemProvider } from "../../src/workspace/file-system-provider";
+import { setFileSystemProvider } from "../../src/workspace/file-system-provider";
+import { startLanguageServer } from "../../src/language-server/connection-handler";
+import { Location } from "vscode-languageserver-types";
+
+describe("Go To Definition request", () => {
+  let fs: VirtualFileSystemProvider;
+  let definitionHandler: (params: {
+    textDocument: { uri: string };
+    position: { line: number; character: number };
+  }) => Location[];
+
+  beforeEach(async () => {
+    fs = new VirtualFileSystemProvider();
+    setFileSystemProvider(fs);
+    TextDocuments.all().forEach((doc) => TextDocuments.delete(doc.uri));
+
+    // Create a mock connection with just the handler we need
+    const mockConnection = {
+      onDefinition: (handler: any) => {
+        definitionHandler = handler;
+      },
+      onInitialize: () => {},
+      onDidChangeTextDocument: () => {},
+      onDidChangeWatchedFiles: () => {},
+      onDidCloseTextDocument: () => {},
+      onDidOpenTextDocument: () => {},
+      onDidSaveTextDocument: () => {},
+      onDidChangeConfiguration: () => {},
+      onDidChangeContent: () => {},
+      onDidClose: () => {},
+      onDidOpen: () => {},
+      onDidSave: () => {},
+      onWillSaveTextDocument: () => {},
+      onWillSaveTextDocumentWaitUntil: () => {},
+      onCompletion: () => {},
+      onReferences: () => {},
+      languages: {
+        semanticTokens: {
+          on: () => {},
+        },
+      },
+      onDocumentHighlight: () => {},
+      onDocumentSymbol: () => {},
+      onRenameRequest: () => {},
+      onWorkspaceSymbol: () => {},
+      listen: () => {},
+      dispose: () => {},
+      sendDiagnostics: () => {},
+    };
+
+    startLanguageServer(mockConnection as any);
```

Comment:
Question: Why do we start the language server and then do nothing with the connection?